### ISSUE TITLE
[Bug] MalformedRoundaboutCheck incorrect flagging for various cases 

### ISSUE THREAD

**Issue by vladlemberg** [osmlab/atlas-checks#318]
**Describe the bug**
Malformed roundabout condition "Car navigable ways inside roundabout" flagging following false positive cases:
1. Roundabout doesnt contain any highway=navigable road.
2. Roundabout contains highway=navigable but its an Area
3. Roundabout spatially intersects with highway=navigable but on different layer.  

**To Reproduce**
Run MalformedRoundAboutCheck against following cases:
1. no navigable rd: https://www.openstreetmap.org/way/693509232
2. navigable but area: https://www.openstreetmap.org/way/792091627
3. multilayer: https://www.openstreetmap.org/way/383475883

**Expected behavior**
Do not flag roundabout

**Screenshots & Stack traces**
<img width="516" alt="Screen Shot 2020-06-18 at 9 20 29 AM" src="https://user-images.githubusercontent.com/62575608/85046457-55e2ad00-b145-11ea-8b35-13eb853a7a50.png">
<img width="397" alt="Screen Shot 2020-06-18 at 9 21 56 AM" src="https://user-images.githubusercontent.com/62575608/85046466-5bd88e00-b145-11ea-8d93-d81f4ee7e978.png">
<img width="561" alt="Screen Shot 2020-06-18 at 9 22 23 AM" src="https://user-images.githubusercontent.com/62575608/85046474-5ed37e80-b145-11ea-883a-97aae802e416.png">

**Comment by Bentleysb** [comment]
Thanks for presenting these case. It looks like there is some enhancement required for the check. Below I have some observations about each of the cases. 
I agree completely that case three is a problem. Clearly the check is not properly accounting for level differences. 
Case two I think is more questionable. I think there is an argument that could be made that the service area should either not overlap the roundabout, or overlap it completely. What I think it comes down to is how routing should be handled through service areas. If routing should be ignored then I think you are correct that this is a false positive. If you should be able to route from any one point to another on the boundary of the area, then I think this is a true positive case. I am not sure what, if any, standards there may be in place about this sort of routing. 
Case three I believe is a true positive. It can be a little hard to see on the OSM tiles, but there is a service road that crosses the roundabout. The service road is not properly connected to the roundabout, and does not appear to exist in satellite imagery. 
Let me know what you think about my observations, especially for case two. As I said there is definitely cases for enhancement here.

**Comment by vladlemberg** [comment]
Thanks for your comments! Agree, Case1,3 required enchantments.  Case2: I believe combination Highway="*", Area=Yes is not mapped to any Functional Road Class. Should be not navigable.

**Comment by vladlemberg** [comment]
@Bentleysb @sayas01 Hello, im working on fixing this issue. 
I'm seeking your opinion for case 1 **"This roundabout has car navigable ways inside it".**  
When i look at the OSM data im not seeing anything wrong:
<img width="511" alt="Screen Shot 2020-07-14 at 1 01 19 PM" src="https://user-images.githubusercontent.com/62575608/87470814-30c14d00-c5d2-11ea-9691-763b767d80a9.png">
Clear evidence that **no car navigable ways inside roundabout.**  

However after debugging
<img width="675" alt="Screen Shot 2020-07-14 at 12 54 06 PM" src="https://user-images.githubusercontent.com/62575608/87471069-97466b00-c5d2-11ea-9715-5c8bbc872b40.png">
I can see that tiny part of conjunction way is actually inside the roundabout. I believe that occurs when editing tool is merging circular and linear shapes. 

Do we want to expand the logic to cover such cases or leave it as is ? For the editors such errors could be treated as False Positive since they are invisible even in JOSM.
<img width="651" alt="Screen Shot 2020-07-14 at 1 11 40 PM" src="https://user-images.githubusercontent.com/62575608/87471684-a11c9e00-c5d3-11ea-94bb-180b707d713c.png">

any thoughts ?  thank you !

**Comment by sayas01** [comment]
@vladlemberg, I discussed with @Bentleysb on this and both of us agree that these cases need to be flagged with clear instructions on what the issue is and adding the locations/points(1 or 2) where the overlap happens.

**Comment by vladlemberg** [comment]
@sayas01 , @Bentleysb, i submitted PR for this. Some comments: 
1) Case #2 Area - fixed
2) Case #3 Multilayer support. Its actually fully handled in **ignoreBridgeTunnelCrossings** method. The reason I got this case during my analysis is that Atlas Data source was outdated. I've created Unit Tests handling all possible conditions.
3) I decided to leave it as is. Adding XY location to crossing point will trigger refactoring in Atlas code base since "enclosed" logic calculating over there.
